### PR TITLE
MGMT-17708: Adjust the logs to show values when registering the cluster

### PR DESCRIPTION
**Pull request by linoyaslan**
Currently, during cluster registration, the new cluster parameters display addresses instead of values:
`log.Infof("Register cluster: %s with id %s and params %+v", swag.StringValue(params.NewClusterParams.Name), id, params.NewClusterParams)`

To resolve this issue, I've modified the logging behavior. Now, the log shows values instead of addresses by marshaling the struct to JSON. Additionally, I've masked the pull-secret value with "***" to protect sensitive information from being exposed in the logs.

## List all the issues related to this PR

- [ ] New Feature 
- [ ] Enhancement 
- [X] Bug fix
- [ ] Tests
- [ ] Documentation
- [ ] CI/CD 

## What environments does this code impact?

- [x] Automation (CI, tools, etc)
- [x] Cloud
- [x] Operator Managed Deployments
- [ ] None

## How was this code tested?

- [X] assisted-test-infra environment
- [ ] dev-scripts environment
- [ ] Reviewer's test appreciated
- [ ] Waiting for CI to do a full test run
- [ ] Manual (Elaborate on how it was tested)
- [ ] No tests needed

## Checklist

- [X] Title and description added to both, commit and PR.
- [X] Relevant issues have been associated (see [CONTRIBUTING] guide)
- [X] This change does not require a documentation update (docstring, `docs`, README, etc)
- [ ] Does this change include unit-tests (note that code changes require unit-tests)

## Reviewers Checklist

- Are the title and description (in both PR and commit) meaningful and clear?
- Is there a bug required (and linked) for this change?
- Should this PR be backported?

[Kubernetes community documentation]: https://github.com/kubernetes/community/blob/master/contributors/guide/pull-requests.md#commit-message-guidelines
[CONTRIBUTING]: https://github.com/openshift/assisted-service/blob/master/CONTRIBUTING.md

### CODE REVIEW

**Comment by danmanor** [comment]
@linoyaslan I think all of these are affected:

- [ ]  Automation (CI, tools, etc)
- [ ]  Cloud
- [ ]  Operator Managed Deployments

File: internal/bminventory/inventory.go [hunk 1]
```diff
@@ -491,10 +491,22 @@ func (b *bareMetalInventory) RegisterClusterInternal(
...
+	// Masked the Pull Secret to prevent its value from being displayed in the logs.
+	clusterParamsNoPullSecret := *params.NewClusterParams
+	if clusterParamsNoPullSecret.PullSecret != nil {
+		*clusterParamsNoPullSecret.PullSecret = "***"
+	}
+
+	jsonNewClusterParams, err := json.Marshal(clusterParamsNoPullSecret)
+	if err != nil {
+		return nil, err
+	}
```
Comment (eliorerz) [inline internal/bminventory/inventory.go:504]: I think it is better to move this code to external method. This function is already long enough

File: internal/bminventory/inventory.go [hunk 2]
```diff
@@ -491,10 +491,22 @@ func (b *bareMetalInventory) RegisterClusterInternal(
...
+
+	log.Infof("Register cluster: %s with id %s and params %s", swag.StringValue(params.NewClusterParams.Name), id, jsonNewClusterParams)
 	success := false
 	cluster := &common.Cluster{}
-	var err error
```
Comment (danmanor) [inline internal/bminventory/inventory.go:497]: why ?

Comment (linoyaslan) [reply internal/bminventory/inventory.go:497]: @danmanor bc line 501

File: internal/bminventory/inventory.go [hunk 3]
```diff
@@ -491,10 +491,22 @@ func (b *bareMetalInventory) RegisterClusterInternal(
 	url := installer.V2GetClusterURL{ClusterID: id}
 
 	log := logutil.FromContext(ctx, b.log).WithField(ctxparams.ClusterId, id)
-	log.Infof("Register cluster: %s with id %s and params %+v", swag.StringValue(params.NewClusterParams.Name), id, params.NewClusterParams)
+
+	// Masked the Pull Secret to prevent its value from being displayed in the logs.
+	clusterParamsNoPullSecret := *params.NewClusterParams
+	if clusterParamsNoPullSecret.PullSecret != nil {
+		*clusterParamsNoPullSecret.PullSecret = "***"
+	}
```
Comment (eliorerz) [inline internal/bminventory/inventory.go:499]: What have we done until now? It was displayed on the logs?

**Review by eliorerz — CHANGES_REQUESTED** [review]

File: internal/bminventory/inventory.go (hunk 3 above)
Comment (linoyaslan) [reply internal/bminventory/inventory.go:499]: @eliorerz The address was displayed

File: internal/bminventory/inventory.go (hunk 2 above)
Comment (danmanor) [reply internal/bminventory/inventory.go:497]: can you send it here please ?

Comment (linoyaslan) [reply internal/bminventory/inventory.go:497]: this line : `jsonNewClusterParams, err := json.Marshal(clusterParamsNoPullSecret)`
But I'm going to move my code to external method according to Elior's suggesion

File: internal/bminventory/inventory.go (hunk 1 above)
Comment (linoyaslan) [reply internal/bminventory/inventory.go]: Due to the error that requires handling, I can't execute it within the log statement, I must execute it beforehand to handle the error, So, I'm not sure if it's worth moving to an external function - what do you think?

**Comment by eliorerz** [comment]
Please add to the PR description the log line you wanted to change

**Review by danmanor — APPROVED** [review]

**Pull request merged**